libs/vm-modules/src/ml/model/model_estimator.cpp

```diff
@@ -216,20 +216,95 @@ ChargeAmount ModelEstimator::CompileSequential(Ptr<String> const &loss,
          COMPUTE_CHARGE_COST;
 }
 
+ChargeAmount ModelEstimator::CompileSequentialWithMetrics(
```

Comment:
you also need to add a couple of estimator tests (see vm-modules/test/unit/model/model_estimator); these should be sanity checks that just ensure the model estimator does what we expect it to over a range of input values